tests/index.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+import sinon from 'sinon';
+import test from 'tape';
+import {
+  Status,
+  callAPI
+} from '../build';
+
+// Global mock payload
+const payload = {
+  foo: 'bar'
+};
+
+// Mock api call as promise with external control
+function createMockAPICallPromise () {
+  var resolve, reject;
+  var mockAPICall = new Promise((resolver, rejecter) => {
+    resolve = resolver;
+    reject = rejecter;
+  });
+
+  return {
+    resolve,
+    reject,
+    mockAPICall
+  };
+};
+
+// Mock action using mock api call
+function createMockActionWithPromise (mockAPICall, shouldCallAPI = true) {
+  return {
+    type: 'MOCK_ACTION',
+    shouldCallAPI: () => shouldCallAPI,
+    callAPI: () => mockAPICall,
+    payload
+  };
+};
+
+// Mock store containing our disaptch spy
+function createMockStore () {
+  const mockStoreState = {
+    foo: 'fiz',
+    bar: 'baz'
+  };
+  var dispatch = sinon.spy();
+
+  return {
+    dispatch,
+    getState: () => mockStoreState
+  };
+};
+
+/*
+ * SUCCESSFUL CALL
+ */
+test('Checking a successful API call', (t) => {
+  var mockPromise = createMockAPICallPromise();
```

Comment:
`deferred` would be a more appropriate name than `promise`. This follows the idea of `Q`'s `deferred` pattern. `createMockAPICallPromise()` returns an object, which contains a promise, but it doesn't return a promise itself. 


```js
var deferred = createMockAPICallPromise()

...

createMockActionWithPromise(deferred.propmise)

...

deferred.resolve('foo')

...
```